### ISSUE TITLE
Salt error in heapster-controller template

### ISSUE THREAD

**Issue by justinsb** [kubernetes/kubernetes#19318]
I observed the following error on AWS when running `salt-call -l debug state.highstate` on the master, on a cluster where I had set `NUM_NODES=30`:

```
[DEBUG   ] Jinja search path: ['/var/cache/salt/minion/files/base']
[INFO    ] Fetching file from saltenv 'base', ** skipped ** latest already in cache 'salt://kube-addons/cluster-monitoring/influxdb/heapster-controller.yaml'
[DEBUG   ] Jinja search path: ['/var/cache/salt/minion/files/base']
[DEBUG   ] Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/salt/states/file.py", line 1336, in managed
    **kwargs
  File "/usr/lib/python2.7/dist-packages/salt/modules/file.py", line 2513, in get_managed
    **kwargs)
  File "/usr/lib/python2.7/dist-packages/salt/utils/templates.py", line 99, in render_tmpl
    output = render_str(tmplstr, context, tmplpath)
  File "/usr/lib/python2.7/dist-packages/salt/utils/templates.py", line 326, in render_jinja_tmpl
    trace=tracestr)
SaltRenderError: Jinja error: unsupported operand type(s) for *: 'NoneType' and 'int'
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/salt/utils/templates.py", line 286, in render_jinja_tmpl
    output = template.render(**unicode_context)
  File "/usr/lib/python2.7/dist-packages/jinja2/environment.py", line 969, in render
    return self.environment.handle_exception(exc_info, True)
  File "/usr/lib/python2.7/dist-packages/jinja2/environment.py", line 742, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "<template>", line 3, in top-level template code
TypeError: unsupported operand type(s) for *: 'NoneType' and 'int'

; line 3

---
{% set heapster_memory = "300Mi" -%}
{% if pillar['num_nodes'] is defined -%}
  {% set heapster_memory = (200 + pillar['num_nodes'] * 12)|string + "Mi" -%}    <======================
{% endif -%}

apiVersion: v1
kind: ReplicationController
metadata:
[...]
---

```

**Comment by justinsb** [comment]
Partial fix in #19320.  But the problem was that num_nodes in pillar was empty:

```> cat /srv/salt-overlay/pillar/cluster-params.sls
instance_prefix: 'kubernetes'
node_instance_prefix: 'kubernetes-minion'
cluster_cidr: '10.244.0.0/16'
allocate_node_cidrs: 'true'
service_cluster_ip_range: '10.0.0.0/16'
enable_cluster_monitoring: 'influxdb'
enable_cluster_logging: 'true'
enable_cluster_ui: 'true'
enable_node_logging: 'true'
logging_destination: 'elasticsearch'
elasticsearch_replicas: '1'
enable_cluster_dns: 'true'
dns_replicas: '1'
dns_server: '10.0.0.10'
dns_domain: 'cluster.local'
admission_control: 'NamespaceLifecycle,LimitRanger,SecurityContextDeny,ServiceAccount,ResourceQuota,PersistentVolumeLabel'
network_provider: 'none'
opencontrail_tag: 'R2.20'
opencontrail_kubernetes_tag: 'master'
opencontrail_public_subnet: '10.1.0.0/16'
num_nodes:
e2e_storage_test_environment: 'false'
```

I think this means that the `{% if pillar['num_nodes'] is defined -%}` in the salt template is not correct.

**Comment by davidopp** [comment]
@brendandburns can you take a look at this? I guess the problem was introduced in #16071
